src/app.jsx

```diff
@@ -38,11 +38,49 @@ export const Application = () => {
             setCurrentUser(user.name || "");
         })
                 .then(() => {
+                    let result = [];
                     const currentPath = path.join("/");
-                    cockpit.spawn(["ls", "-p", `/home/${currentUser}/${currentPath}`], { superuser: true }).then((res) => {
-                        res = res.split("\n");
-                        setFiles(res.slice(0, res.length - 1));
+                    const channel = cockpit.channel({
+                        payload: "fslist1",
+                        path: `/home/${currentUser}/${currentPath}`,
+                        superuser: "try",
+                        watch: true,
                     });
+
+                    channel.addEventListener("ready", () => {
+                        setFiles(result);
+                        console.log("ready");
+                    });
+
+                    channel.addEventListener("close", (ev, data) => {
+                        console.log("close");
+                    });
+
+                    channel.addEventListener("message", (ev, data) => {
+                        const item = JSON.parse(data);
+                        console.log(item);
+                        // setFiles([...files, { name: item.path, type: item.type }]);
+                        if (item && item.path && item.event === 'present' && item.path[0] !== ".")
+                            result.push({ name: item.path, type: item.type });
+                        else if (item.event === 'deleted') {
+                            const deleted_name = item.path.slice(item.path.lastIndexOf("/") + 1);
+                            console.log(deleted_name);
+                            result = result.filter((res) => { return res.name !== deleted_name });
+                            setFiles(result);
+                            console.log(files);
+                        } else if (item.event === 'created') {
+                            const created_name = item.path.slice(item.path.lastIndexOf("/") + 1);
+                            console.log(created_name);
+                            result.push({ name: created_name, type: item.type });
+                            console.log(result);
+                            setFiles(result);
```

Comment:
You can't do this. `result` is a long-lived variable, which you assign as React state. Whenever you write to it (like `result.push()` does), you change the *existing* state "through the backdoor", which is buggy. Please see https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state

Reply:
That link contained exactly what I was looking for!
Using an updater function fixed the issue here, I will keep it in mind for the future :)